core/identity-hub-client/src/main/java/org/eclipse/edc/identityhub/client/IdentityHubClientImpl.java

```diff
@@ -133,16 +138,42 @@ public StatusResult<Void> addVerifiableCredential(String hubBaseUrl, SignedJWT v
             if (response.code() != 200) {
                 return identityHubCallError(response);
             }
+            var responseObject = objectMapper.readValue(response.body().byteStream(), ResponseObject.class);
+
+            if (responseObject.getStatus() != null && !responseObject.getStatus().isSuccess()) {
+                return StatusResult.failure(ResponseStatus.FATAL_ERROR, responseObject.getStatus().getDetail());
+            }
+
+            var requestStatus = Optional.of(responseObject.getStatus())
+                    .map(status -> Stream.of(status))
+                    .orElseGet(Stream::empty);
+
+            var messageStatus = responseObject.getReplies()
+                    .stream()
+                    .map(MessageResponseObject::getStatus);
+
+            return Stream.concat(messageStatus, requestStatus)
```

Comment:
The logic in this new block of code lost me... :)
Perhaps would be better to encapsulate into a dedicated method documenting the behavior.

Reply:
Yeah i can move the result handling in a separated method with doc.